src/api/app/controllers/webui/badge_controller.rb

```diff
@@ -0,0 +1,39 @@
+class Webui::BadgeController < Webui::WebuiController
+  before_action :set_project
+  before_action :require_package
+
+  def show
+    results = buildresults
+    results = results.select { |r| r.architecture == params[:arch] } if params[:arch]
+    results = results.select { |r| r.repository == params[:repo] } if params[:repo]
+    send_data(create_badge(results), type: 'image/svg+xml', disposition: 'inline')
+  end
+
+  private
+
+  def buildresults
+    Rails.cache.fetch(['buildresults', @project.name, @package.name], expires_in: 10.minutes) do
```

Comment:
I'm not sure a time based cache invalidation is worth it. We allow fetching build results anonymously from `Webui::PackageController#show`, `Webui::RequestController#show` and `Webui::ProjectController#monitor` and possibly more places. An event based  cache invalidation would be nice, but you don't have to do this now.